### PR TITLE
Only specify CARGO_TARGET_..._LINKER if not already specified, and ensure it and `STD_TARGETS` are exported and avaliable to `./run`.

### PR DESCRIPTION
**Pull request by thomcc**
Reported on discord.

### CODE REVIEW

File: plrust/build [hunk 1]
```diff
@@ -3,17 +3,21 @@ set -xe
 
 if [ -z "$STD_TARGETS" ]; then
     # if none specified, always build for these two targets
-    STD_TARGETS="x86_64-postgres-linux-gnu aarch64-postgres-linux-gnu"
+    export STD_TARGETS="x86_64-postgres-linux-gnu aarch64-postgres-linux-gnu"
```
Comment (workingjubilee) [inline plrust/build:6]: This should instead be explicitly passed to `./run install`, to indicate the dependency.

Comment (workingjubilee) [reply plrust/build:6]: This was how it historically was, but it was undone as part of #165 changes.

Comment (thomcc) [reply plrust/build]: It needs to be passed to clean too, now that cross compilation exists.

File: plrust/build [hunk 2]
```diff
@@ -46,8 +46,8 @@ fi
         cd ./postgrestd
     fi
     rm --force rust-toolchain.toml
-    ./run clean
-    ./run install
+    STD_TARGETS="$STD_TARGETS" ./run clean
```
Comment (workingjubilee) [inline plrust/build:49]: I'm not sure why this acquired a `./run clean` given that `./run install` already does what should be enough but I am gonna figure that out later.

**Review by workingjubilee — APPROVED** [review]
Nice, should be good!

**Pull request merged**